From 8afd705ea7ddffd363edfd443457597c878928a9 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:44 -0400
Subject: [infiniband] cxgb4: Cap CQ size at T4_MAX_IQ_SIZE

Message-id: <1410475447-94392-26-git-send-email-svemuri@redhat.com>
Patchwork-id: 93795
O-Subject: [RHEL7.1 PATCH BZ 1124947 025/108] RDMA/cxgb4: Cap CQ size at T4_MAX_IQ_SIZE
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit ffd435924c86de055d33fe59941841819eef9f6a)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/cq.c b/drivers/infiniband/hw/cxgb4/cq.c
index e436ead..906119f 100644
--- a/drivers/infiniband/hw/cxgb4/cq.c
+++ b/drivers/infiniband/hw/cxgb4/cq.c
@@ -881,7 +881,7 @@ struct ib_cq *c4iw_create_cq(struct ib_device *ibdev, int entries,
  /*
   * Make actual HW queue 2x to avoid cdix_inc overflows.
   */
- hwentries = entries * 2;
+ hwentries = min(entries * 2, T4_MAX_IQ_SIZE);
 
  /*
   * Make HW queue at least 64 entries so GTS updates aren't too
-- 
1.7.1